### PR TITLE
refactor(cmd_all): refactor playground to use `single_node --in-memory` internally

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Remove the `playground` internals. `playground` will just be an alias of:
```
./risingave --in-memory --max-idle-secs=1800
```

Note that previously we have other profiles like:
- `playground-3cn`
- `online-docker-playground`

But these are removed. Only the `playground` (default) profile is left.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by fuyufjh** [comment]
I tend to remove `playground` completely and use the standard `./risingwave` in playground.risingwave.com

@mikechesterwang Can you please add these env vars in for web playground container:

```
RW_SINGLE_NODE_MAX_IDLE_SECS = 1800
// RW_SINGLE_NODE_IN_MEMORY = true -- I think this is not needed any more
```

and change the entrypoint command to `./risingwave`?

**Comment by kwannoel** [comment]
> I tend to remove `playground` completely and use the standard `./risingwave` in playground.risingwave.dev
> 
> @mikechesterwang Can you please add these env vars in for web playground container:
> 
> ```
> RW_SINGLE_NODE_MAX_IDLE_SECS = 1800
> // RW_SINGLE_NODE_IN_MEMORY = true -- I think this is not needed any more
> ```
> 
> and change the entrypoint command to `./risingwave`?

The alias `./risingwave playground` is for backwards compatibility for users.

**Comment by fuyufjh** [comment]
> The alias `./risingwave playground` is for backwards compatibility for users.

Hmm, regarding of "playground", I don't think we need to preserve backwards compatibility

**Comment by kwannoel** [comment]
> > The alias `./risingwave playground` is for backwards compatibility for users.
> 
> Hmm, regarding of "playground", I don't think we need to preserve backwards compatibility

To me, it's not a big deal either ways.
1. It's not a lot of code to maintain `./risingwave playground`.
2. Or, if we remove it, users can just use ./risingwave --in-memory as a replacement.

I'll let @BugenZhao comment more.

File: src/cmd_all/src/single_node.rs [hunk 1]
```diff
@@ -53,6 +53,16 @@ pub struct SingleNodeOpts {
     node_opts: NodeSpecificOpts,
 }
 
+impl SingleNodeOpts {
+    pub fn new_for_playground() -> Self {
+        let empty_args = vec![] as Vec<String>;
+        let mut opts = SingleNodeOpts::parse_from(empty_args);
+        opts.in_memory = true;
+        opts.max_idle_secs = Some(1800);
```
Comment (BugenZhao) [inline src/cmd_all/src/single_node.rs:61]:
```suggestion
```

...as this was originally for the online playground and it's going to switch to `--in-memory`. Our developers were not fans of this.

File: src/cmd_all/src/playground.rs [hunk 2]
```diff
@@ -1,260 +0,0 @@
...
-    }
-
-    // wait for log messages to be flushed
-    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
-    eprintln!("-------------------------------");
-    eprintln!("RisingWave playground is ready.");
-    eprint!(
-        "* {} RisingWave playground SHOULD NEVER be used in benchmarks and production environment!!!\n  It is fully in-memory",
-        console::style("WARNING:").red().bold(),
-    );
```
Comment (BugenZhao) [inline src/cmd_all/src/playground.rs:216]: Can we keep these prompts and caveats?

**Review by BugenZhao — APPROVED** [review]
Rest LGTM

File: src/cmd_all/src/single_node.rs (hunk 1 above)
Comment (kwannoel) [reply src/cmd_all/src/single_node.rs:61]: Sounds good. It's an alias for internal dev then... I will just `hide` it from top-level if possible...

For cloud playground, we can use `./risingwave --in-memory --max-idle-secs=1800` to invoke instead.

**Comment by BugenZhao** [comment]
To be honest, I still don't find any motivation to deprecate the entrypoint of `risingwave playground`. 🤡 I'm a big fan of launching `risingwave playground` locally to develop and debug.

Adopting `--in-memory` in the online playground does not have anything to do with using `playground` for local development IMO.

**Pull request merged**